libsolidity/ast/Types.cpp

```diff
@@ -307,7 +331,7 @@ string IntegerType::toString(bool) const
 
 TypePointer IntegerType::binaryOperatorResult(Token::Value _operator, TypePointer const& _other) const
 {
-	if (_other->category() != Category::IntegerConstant && _other->category() != category())
+	if (_other->category() != Category::RationalNumber && _other->category() != category())
 		return TypePointer();
 	auto commonType = dynamic_pointer_cast<IntegerType const>(Type::commonType(shared_from_this(), _other));
```

Comment:
The conditions below are not 100% correct. It will allow the following:
```
uint8 x = 2;
ufixed y = .5;
x**y
```
Please make a note somewhere to add a test for that once we get to implementing fixed point types themselves.